ClosedXML/Excel/Cells/XLCells.cs

```diff
@@ -71,6 +71,22 @@ private IEnumerable<XLSheetPoint> GetAllCellsInRange(IXLRangeAddress rangeAddres
 
         private IEnumerable<XLCell> GetUsedCells()
         {
+            if (_rangeAddresses.Count == 1)
+            {
+                // fast path
+                var ws = _rangeAddresses[0].Worksheet;
+                var usedCellsCandidates = GetUsedCellsCandidates(ws);
+
+                return GetUsedCellsInRange(_rangeAddresses[0], ws, usedCellsCandidates);
+            }
+
+            // need to do a bit more complex aggregation
+            return GetUsedCellsByWorksheetGroup();
+        }
+
+        private IEnumerable<XLCell> GetUsedCellsByWorksheetGroup()
+        {
+            var visitedCells = new HashSet<XLAddress>();
```

Comment:
reusing same HashSet instead of always creating a new one in loop, there were a lot of them created even when one can be used if cleared between iterations